packages/snap/src/snap-test/set-batch-id.ts

```diff
@@ -0,0 +1,30 @@
+import {execSync} from 'child_process';
+
+const PULL_REQUEST_PARENT_HASH_INDEX = 2;
+const HEAD_HASH_INDEX = 0;
+
+function getHeadHash() {
+  return execSync('git rev-parse --verify HEAD').toString();
+}
+
+function getParentsHashArray() {
+  const headCommitHash = getHeadHash();
+  return execSync(`git rev-list --parents -n 1 ${headCommitHash}`).toString().split(' ');
+}
+
+function getPRHeadHash() {
```

Comment:
I think the name of the function is confusing, since it checks for `isPullRequest` it doesn't always return the PR head hash, the question then is what it returns. I don't want to have any magical code that we don't understand in the project.

Reply:
Added warning regarding the magical quality of said method